usr/share/rear/rescue/GNU/Linux/310_network_devices.sh

```diff
@@ -101,16 +104,73 @@ for network_interface in $network_interfaces ; do
     # virtual_network_interfaces is "lo virt-eth0" and network_interfaces is "eth0 lo virt-eth0"
     # then simple substring search would find "eth0" as substring in "lo virt-eth0"
     # so that to be on the safe side a dumb traditional for-loop approach is used
-    # (unitl someone implements a better solution that works on all bash 3.x versions):
-    for virtual_network_interface in $virtual_network_interfaces ; do
-        test "$network_interface" = "$virtual_network_interface" && network_interface=""
-    done
+    # (until someone implements a better solution that works on all bash 3.x versions):
+    if [ -d "/sys/class/net/$network_interface/bridge" ]; then
+        # Consider bridges linked to physical interfaces only (directly or not)
+        function is_linked_to_physical () {
+            local devpath=$1
+            # If device is a physical device, return success
+            [ ! -e $devpath/device ] || return 0
+            # Otherwise recurse on lower devices (if any)
+            for dev in $(/bin/readlink -f $devpath/lower_* 2>/dev/null); do
+                ! is_linked_to_physical $dev || return 0
+            done
+            # Device is not a physical device
+            return 1
+        }
+        is_linked_to_physical /sys/class/net/$network_interface || network_interface=""
```

Comment:
Like the other local functions also the
"very local" function is_linked_to_physical should be unset
at the end of the scope where it is defined and used
(i.e. at the end of that 'if' condition body)
<pre>
if ...
    function is_linked_to_physical () {
        ....
    }
    is_linked_to_physical ...
    unset -f is_linked_to_physical
else
</pre>